picard/oauth.py

```diff
@@ -44,41 +44,99 @@ class OAuthManager(object):
+
+    @refresh_token_scopes.deleter
+    def refresh_token_scopes(self):
+        self.persist.remove("oauth_refresh_token_scopes")
+
+    @property
+    def access_token(self):
+        return self.persist["oauth_access_token"]
+
+    @access_token.setter
+    def access_token(self, value):
+        self.persist["oauth_access_token"] = value
+
+    @access_token.deleter
+    def access_token(self):
+        self.persist.remove("oauth_access_token")
+
+    @property
+    def access_token_expires(self):
+        return self.persist["oauth_access_token_expires"]
+
+    @access_token_expires.setter
+    def access_token_expires(self, value):
+        self.persist["oauth_access_token_expires"] = value
+
+    @access_token_expires.deleter
+    def access_token_expires(self):
+        self.persist.remove("oauth_access_token_expires")
+
+    @property
+    def username(self):
+        return self.persist["oauth_username"]
+
+    @username.setter
+    def username(self, value):
+        self.persist["oauth_username"] = value
 
     def is_authorized(self):
-        return bool(config.persist["oauth_refresh_token"] and
-                    config.persist["oauth_refresh_token_scopes"])
+        return bool(self.refresh_token and self.refresh_token_scopes)
 
     def is_logged_in(self):
-        return self.is_authorized() and bool(config.persist["oauth_username"])
+        return self.is_authorized() and bool(self.username)
 
     def revoke_tokens(self):
         # TODO actually revoke the tokens on MB (I think it's not implementented there)
         self.forget_refresh_token()
         self.forget_access_token()
 
     def forget_refresh_token(self):
-        config.persist.remove("oauth_refresh_token")
-        config.persist.remove("oauth_refresh_token_scopes")
+        del self.refresh_token
+        del self.refresh_token_scopes
 
     def forget_access_token(self):
-        config.persist.remove("oauth_access_token")
-        config.persist.remove("oauth_access_token_expires")
+        del self.access_token
+        del self.access_token_expires
 
     def get_access_token(self, callback):
         if not self.is_authorized():
             callback(None)
         else:
-            access_token = config.persist["oauth_access_token"]
-            access_token_expires = config.persist["oauth_access_token_expires"]
-            if access_token and time.time() < access_token_expires:
-                callback(access_token)
+            if self.access_token and time.time() < self.access_token_expires:
+                callback(self.access_token)
             else:
                 self.forget_access_token()
                 self.refresh_access_token(callback)
 
+    @staticmethod
+    def host_port():
+        return config.setting['server_host'], config.setting['server_port']
+
     def get_authorization_url(self, scopes):
-        host, port = config.setting['server_host'], config.setting['server_port']
+        host, port = self.host_port()
```

Comment:
Why not make these properties as you've done for other settings?

Reply:
You're right, fixed in 4269fb8e8ea74cd2acdb603dde4bc7254557b826